<launch>
    <node pkg="object_detection" type="object_detection_node" name="object_detection_node">
        <!-- input topic name -->
        <param name="point_cloud_topic_name" type="string" value="/pointcloud/os1"/>
        <param name="camera_rsi_topic_name" type="string" value="/vds_node_localhost_2211/image_raw/compressed"/>
        <param name="cmnode_topic_name" type="string" value="/hellocm/cm2ext"/>

        <!-- output topic name-->
        <param name="object_topic_name" type="string" value="/object_detection/objects"/>

        <!-- raw Lidar sensor paramaters -->
        <param name="lidar_rsi_position_m_x" type="double" value="2.4"/>
        <param name="lidar_rsi_position_m_y" type="double" value="0.0"/>
        <param name="lidar_rsi_position_m_z" type="double" value="2.2"/>

        <param name="lidar_rsi_rotation_deg_x" type="double" value="0.0"/>
        <param name="lidar_rsi_rotation_deg_y" type="double" value="0.0"/>
        <param name="lidar_rsi_rotation_deg_z" type="double" value="0.0"/>

        <param name="segment_distance_threshold" type="double" value="0.35"/>
        <param name="roi_range_x" type="double" value="40.0"/>
        <param name="roi_range_y" type="double" value="15.0"/>
        <param name="roi_range_z" type="double" value="4.0"/>

        <param name="cluster_tolerance" type="double" value="0.7" />
        <param name="min_cluster_size" type="int" value="10" />
        <param name="max_cluster_size" type="int" value="5000" />

        <!-- raw camera sensor parameters -->
        <param name="focal_length_x" type="double" value="381.3611496"/>
        <param name="focal_length_y" type="double" value="381.3611496"/>
        <param name="focal_center_x" type="double" value="320"/>
        <param name="focal_center_y" type="double" value="240"/>

        <param name="camera_rsi_position_m_x" type="double" value="4.0"/>
        <param name="camera_rsi_position_m_y" type="double" value="0.0"/>
        <param name="camera_rsi_position_m_z" type="double" value="1.5"/>

        <param name="camera_rsi_rotation_deg_x" type="double" value="-90.0"/>
        <param name="camera_rsi_rotation_deg_y" type="double" value="0.0"/>
        <param name="camera_rsi_rotation_deg_z" type="double" value="-90.0"/>

        <!-- image processing output parameters (camera GT) -->
        <param name="camera_gt_position_m_x" type="double" value="2.5"/>
        <param name="camera_gt_position_m_y" type="double" value="0.0"/>
        <param name="camera_gt_position_m_z" type="double" value="1.3"/>

        <param name="camera_gt_rotation_deg_x" type="double" value="0.0"/>
        <param name="camera_gt_rotation_deg_y" type="double" value="0.0"/>
        <param name="camera_gt_rotation_deg_z" type="double" value="0.0"/>

    </node>
</launch>